lib/std/setutils.nim

```diff
@@ -0,0 +1,33 @@
+#
+#
+#           The Nim Compiler
+#        (c) Copyright 2020 Nim Contributors
+#
+#    See the file "copying.txt", included in this
+#    distribution, for details about the copyright.
+#
+
+## This module adds functionality to the default set
+
+import typetraits
+
+type SetElement* = char|byte|bool|int16|uint16|enum|uint8|int8
+
+template toSet*(iter: untyped): untyped =
+  ##Iterates through an openArray making a set from it.
+  runnableExamples: 
```

Comment:
this module also needs a `tests/stdlib/tsetutils.nim` where could can test more things